src/features/governance/fetchHistoricalGovernanceEvents.ts

```diff
@@ -0,0 +1,116 @@
+/* eslint no-console: 0 */
+
+import { governanceABI } from '@celo/abis';
+import { resolveAddress } from '@celo/actions';
+import { sql } from 'drizzle-orm';
+import database from 'src/config/database';
+import { blocksProcessedTable, eventsTable } from 'src/db/schema';
+import {
+  Chain,
+  GetContractEventsParameters,
+  HttpRequestError,
+  PublicClient,
+  RpcRequestError,
+  TimeoutError,
+  Transport,
+} from 'viem';
+
+import '../../vendor/polyfill.js';
+
+const default_step = 100_000n;
+
+const bigintMath = {
+  min: (...args: bigint[]) => args.reduce((min_, x) => (x < min_ ? x : min_)),
+  max: (...args: bigint[]) => args.reduce((max_, x) => (x > max_ ? x : max_)),
+};
+
+export default async function fetchHistoricalEvents(
```

Comment:
this does a lot more than fetching

fetchHistoricalEvents
UpsertHistoricalEvents
markBlocksProcessed.